tcms/rpc/tests/test_testrun.py

```diff
@@ -312,3 +404,26 @@ def test_wrong_date_format(self):
         self.assertNotEqual(update_fields['summary'], test_run.summary)
         self.assertNotEqual(datetime.strptime(update_fields['stop_date'], '%d-%m-%Y'),
                             test_run.stop_date)
+
+    def test_update_with_product(self):
+        test_run = TestRunFactory()
+        product = ProductFactory()
+        updated_test_plan = TestPlanFactory(product=product)
+        updated_build = BuildFactory(product=product)
+        updated_summary = 'Updated summary.'
+        updated_stop_date = datetime.strptime('2020-05-05', '%Y-%m-%d')
+
+        update_fields = {
+            'plan': updated_test_plan.pk,
+            'product': product.id,
+            'build': updated_build.pk,
+            'summary': updated_summary,
+            'stop_date': updated_stop_date
```

Comment:
This field accepts also string values so it isn't necessary to convert back and forth.

Reply:
good point, i leave this as is and use string value in permission case.

Reply:
as-is is performing unnecessary conversions:
1. from string to datetime on line 437
2. from datatime to string on line 452

Use a string value for `updated_stop_date` and don't convert it back and forth because it is not needed.